### PR TITLE
[trel] enable/disable TREL with feature flag

### PR DESCRIPTION
**Pull request by sunytt**
Use feature flag to enable/disable the TREL feature

### CODE REVIEW

**Review by morningboata — APPROVED** [review]

File: src/ncp/ncp_openthread.cpp [hunk 1]
```diff
@@ -265,6 +266,17 @@ otError ControllerOpenThread::ApplyFeatureFlagList(const FeatureFlagList &aFeatu
         error = SetOtbrAndOtLogLevel(otbrLogGetDefaultLevel());
     }
 
+#if OTBR_ENABLE_TREL
+    if (aFeatureFlagList.enable_trel())
+    {
+        otTrelEnable(mInstance);
```
Comment (morningboata) [inline src/ncp/ncp_openthread.cpp:272]: Is there any error code return from the otTrelEnable API? May need to return the error if any.

**Review by morningboata — APPROVED** [review]

File: src/ncp/ncp_openthread.cpp (hunk 1 above)
Comment (superwhd) [reply src/ncp/ncp_openthread.cpp:272]: `otTrelEnable` returns void

Comment (superwhd) [inline src/ncp/ncp_openthread.cpp:272]: nit: may provide `otTrelSetEnabled` API in a future PR to simplify the code

**Review by superwhd — APPROVED** [review]
LGTM :+1:

**Review by librasungirl — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**